gcal_sync/model.py

```diff
@@ -790,23 +780,23 @@ def _adjust_unknown_event_type(cls, values: dict[str, Any]) -> dict[str, Any]:
                 values["eventType"] = EventTypeEnum.UNKNOWN
         return values
 
-    @root_validator
-    def _adjust_resource_all_day_event(cls, values: dict[str, Any]) -> dict[str, Any]:
+    @model_validator(mode="after")
+    def _adjust_resource_all_day_event(self) -> Self:
         """Fix a bug in google calendar resources where all day events are incorrect."""
         if (
-            (calendar_id := values.get("private_calendar_id"))
+            (calendar_id := self.private_calendar_id)
             and calendar_id.endswith(RESOURCE_ID_SUFFIX)
-            and (dtstart := values.get("start"))
-            and (dtend := values.get("end"))
+            and (dtstart := self.start)
+            and (dtend := self.end)
             and dtstart.date_time
-            and dtstart.date_time
-            and dtend.date_time.time() == MIDNIGHT
+            and dtend.date_time
+            and dtstart.date_time.time() == MIDNIGHT
             and dtend.date_time.time() == MIDNIGHT
```

Comment:
Probably a small bug that `dtstart.date_time` and `dtend.date_time.time() == MIDNIGHT` are both checked twice but their respective `dtend.date_time` and `dtstart.date_time.time()` aren't checked at all.